deepmd/entrypoints/test.py

```diff
@@ -933,20 +933,37 @@ def test_dipole(
     log.info("The unit of error is the same as the unit of provided label.")
 
     if detail_file is not None:
-        detail_path = Path(detail_file)
+        if not atomic:
+            detail_path = Path(detail_file)
 
-        pe = np.concatenate(
-            (
-                np.reshape(test_data["dipole"][:numb_test], [-1, 3]),
-                np.reshape(dipole, [-1, 3]),
-            ),
-            axis=1,
-        )
-        np.savetxt(
-            detail_path.with_suffix(".out"),
-            pe,
-            header="data_x data_y data_z pred_x pred_y pred_z",
-        )
+            pe = np.concatenate(
+                (
+                    np.reshape(test_data["dipole"][:numb_test], [-1, 3]),
+                    np.reshape(dipole, [-1, 3]),
+                ),
+                axis=1,
+            )
+
+            np.savetxt(
+                detail_path.with_suffix(".out"),
+                pe,
+                header="data_x data_y data_z pred_x pred_y pred_z",
+            )
+        else:
+            detail_path = Path(detail_file)
+
+            pe = np.concatenate(
+                (
+                    test_data["atomic_dipole"][:numb_test],
+                    dipole,
+                ),
+                axis=1,
+            )
+
+            np.savetxt(
+                detail_path.with_suffix(".out"),
+                pe,
```

Comment:
`header` should be added here, like line 950.

Reply:
It seems that we do not need to write it twice. Just put these lines out of the `if...else`.